daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1909,7 +1909,7 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
     funcObj
   }
 
-  def lookupTypeCalculator(typeCalcExpr: Expression, requiresParse: Boolean, requiresUnparse: Boolean): TypeCalculator[AnyRef, AnyRef] = {
+  def lookupTypeCalculator(typeCalcExpr: Expression, requiresParse: Boolean, requiresUnparse: Boolean): TypeCalculator = {
```

Comment:
We no longer attempt to represent the types a particular calculator uses with the Scala type system. The src/dst types are saved as fields of the calculator. From the perspective of the Scala type system, all calculators are now DataValue -> DataValue